OptimizelySDK.Tests/OptimizelyTest.cs

```diff
@@ -1109,5 +1120,350 @@ public void TestGetVariationBucketingIdAttribute()
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyNonBoolean, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns("non_boolean_value");
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyNull, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns<string>(null);
+
+            Assert.AreEqual(true, OptimizelyMock.Object.GetFeatureVariableBoolean(featureKey, variableKeyTrue, TestUserId, null));
+            Assert.AreEqual(false, OptimizelyMock.Object.GetFeatureVariableBoolean(featureKey, variableKeyFalse, TestUserId, null));
+            Assert.AreEqual(null, OptimizelyMock.Object.GetFeatureVariableBoolean(featureKey, variableKeyNonBoolean, TestUserId, null));
+            Assert.AreEqual(null, OptimizelyMock.Object.GetFeatureVariableBoolean(featureKey, variableKeyNull, TestUserId, null));
+        }
+
+        [Test]
+        public void TestGetFeatureVariableDoubleReturnTypecastedValue()
+        {
+            var featureKey = "featureKey";
+            var variableKeyDouble = "varDouble";
+            var variableKeyInt = "varInt";
+            var variableKeyNonDouble = "varNonDouble";
+            var variableKeyNull = "varNull";
+            var featureVariableType = FeatureVariable.VariableType.DOUBLE;
+
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyDouble, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns("100.54");
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyInt, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns("100");
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyNonDouble, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns("non_double_value");
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyNull, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns<string>(null);
+
+            Assert.AreEqual(100.54, OptimizelyMock.Object.GetFeatureVariableDouble(featureKey, variableKeyDouble, TestUserId, null));
+            Assert.AreEqual(100, OptimizelyMock.Object.GetFeatureVariableDouble(featureKey, variableKeyInt, TestUserId, null));
+            Assert.AreEqual(null, OptimizelyMock.Object.GetFeatureVariableDouble(featureKey, variableKeyNonDouble, TestUserId, null));
+            Assert.AreEqual(null, OptimizelyMock.Object.GetFeatureVariableDouble(featureKey, variableKeyNull, TestUserId, null));
+        }
+
+        [Test]
+        public void TestGetFeatureVariableIntegerReturnTypecastedValue()
+        {
```

Comment:
add log messages.

Reply:
We can't do that. GetFeatureVariable<Type> methods do not log any message inside them. They just call GetFeatureVariableValueForType and we mocked that method to return the desired value.

Reply:
Done